src/components/stepforms/widgets/IfThenElseWidget.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div class="ifthenelse-widget">
+    <div class="ifthenelse-container">
+      <div v-if="isRoot" class="ifthenelse-widget__if">IF:</div>
+      <FilterEditor
+        :filter-tree="editedValue.if"
+        :errors="errors"
+        :data-path="`${dataPath}.if`"
+        @filterTreeUpdated="updateFilterTree"
+      />
+      <div class="ifthenelse-widget__then">THEN:</div>
+      <InputTextWidget
+        class="formulaInput"
+        :value="editedValue.then"
+        @input="updateThenFormula"
+        placeholder='Enter a "Text" with quotes, or a formula'
+        :data-path="`${dataPath}.then`"
+        :errors="errors"
+      />
+      <AutocompleteWidget
+        class="ifthenelse-widget__else"
+        v-model="elseMode"
+        @input="updateElseMode"
+        :options="elseModes"
+      />
+      <InputTextWidget
+        v-if="elseMode === 'ELSE:'"
+        class="formulaInput"
+        :value="editedValue.else"
+        @input="updateElseFormula"
+        placeholder='Enter a "Text" with quotes, or a formula'
+        :data-path="`${dataPath}.else`"
+        :errors="errors"
+      />
+    </div>
+    <ifthenelse-widget
+      v-if="elseMode === 'ELSE IF:'"
+      :isRoot="false"
+      :value="editedValue.else"
+      @input="updateElseObject"
+      :data-path="`${dataPath}.else`"
+      :errors="errors"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { ErrorObject } from 'ajv';
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+import FilterEditor from '@/components/FilterEditor.vue';
+import AutocompleteWidget from '@/components/stepforms/widgets/Autocomplete.vue';
+import InputTextWidget from '@/components/stepforms/widgets/InputText.vue';
+import { ColumnTypeMapping } from '@/lib/dataset';
+import { FilterComboAnd, FilterComboOr, FilterSimpleCondition, IfThenElseStep } from '@/lib/steps';
+import { VQBModule } from '@/store';
+
+@Component({
+  name: 'ifthenelse-widget',
+  components: {
+    AutocompleteWidget,
+    FilterEditor,
+    InputTextWidget,
+  },
+})
+export default class IfThenElseWidget extends Vue {
+  @Prop({
+    type: Object,
+    default: () => ({
+      if: { column: '', value: '', operator: 'eq' },
+      then: '',
+      else: '',
+    }),
+  })
+  value!: Omit<IfThenElseStep, 'name' | 'newColumn'>;
+
+  @Prop({ type: String, default: '' })
+  dataPath!: string;
+
+  @Prop({
+    type: Boolean,
+    default: true,
+  })
+  isRoot!: boolean;
+
+  @Prop({
+    type: Array,
+    default: () => [],
+  })
+  errors!: ErrorObject[];
+
+  @VQBModule.Getter columnTypes!: ColumnTypeMapping;
+
+  readonly title: string = 'Add a conditional column';
+  readonly elseModes = ['ELSE:', 'ELSE IF:'];
+  elseMode = typeof this.value.else === 'string' ? 'ELSE:' : 'ELSE IF:';
+  editedValue = { ...this.value };
```

Comment:
This component `emit` "immediately" the new value (it does not wait for a user action like a "save"), you should not need `editedValue`. Simply pass down `this.value` to child component, and emit an updated value using the `update{stuff}` methods. (In other word: this component does not need an internal state.)

Note: `editedValue` is useful when your component `emit` at a "save" action, like in step, because you cannot mutate the props, you need an internal state.